docs/adr/001-provisioning.md

```diff
@@ -0,0 +1,404 @@
+    name: shoot-name
+    # spec.shoot.purpose is required
+    purpose: production
+    # spec.shoot.region is required
+    region: eu-central-1
+    # spec.shoot.secretBindingName is required
+    secretBindingName: "hyperscaler secret"
+    kubernetes:
+      kubeAPIServer:
+        ## spec.shoot.kubernetes.kubeAPIServer.oidcConfig is required
+        oidcConfig:
+          clientID: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+          groupsClaim: groups
+          issuerURL: https://my.cool.tokens.com
+          signingAlgs:
+            - RS256
+          usernameClaim: sub
+    provider:
+      ## spec.shoot.provider.type is required
+      type: aws
+      # spec.shoot.provider.controlPlaneConfig is required
+      controlPlaneConfig:
+        apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
+        kind: ControlPlaneConfig
+      # spec.shoot.provider.infrastructureConfig is required
+      infrastructureConfig:
+        apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
+        kind: InfrastructureConfig
+        networks:
+          vpc:
+            cidr: 10.250.0.0/16
+          zones:
+            - internal: 10.250.48.0/20
+              name: eu-central-1c
+              public: 10.250.32.0/20
+              workers: 10.250.0.0/19
+            - internal: 10.250.112.0/20
+              name: eu-central-1b
+              public: 10.250.96.0/20
+              workers: 10.250.64.0/19
+            - internal: 10.250.176.0/20
+              name: eu-central-1a
+              public: 10.250.160.0/20
+              workers: 10.250.128.0/19
+    # spec.shoot.Networking is required
+    networking:
+      pods: 100.64.0.0/12
+      nodes: 10.250.0.0/16
+      services: 100.104.0.0/13
+    # spec.shoot.controlPlane is required
+    controlPlane:
+      highAvailability:
+        failureTolerance:
+          type: node
+    workers:
+      - machine:
+          # spec.shoot.workers.machine.type is required
+          type: m6i.large
+        # spec.shoot.workers.zones is required
+        zones:
+          - eu-central-1a
+          - eu-central-1b
+          - eu-central-1c
+        # spec.shoot.workers.minimum is required
+        minimum: 3
+        # spec.shoot.workers.maximum is required
+        maximum: 20
+        # spec.shoot.workers.maxSurge is required in the first release.
+        # It can be optional in the future, as it equals to zone count
+        maxSurge: 3
+        # spec.shoot.workers.maxUnavailable is required in the first release.
+        # It can be optional in the future, as it is always set to 0
+        maxUnavailable:  0
+  security:
+    networking:
+      filter:
+        # spec.security.networking is required
+        egress:
+          enabled: false
+    # spec.security.administrators is required
+    administrators:
+      - admin@myorg.com
+```
+
+There are some additional optional fields that could be specified:
+- `spec.shoot.seedName` ; if not provided `nil` value will be used
```

Comment:
Gardener does not allow setting seedName any more, this is a read-only attribute guarded by gardener-webhook